packages/babel-types/src/definitions/es2015.js

```diff
@@ -42,28 +45,51 @@ defineType("ArrowFunctionExpression", {
   visitor: ["params", "body", "returnType", "typeParameters"],
   aliases: ["Scopable", "Function", "BlockParent", "FunctionParent", "Expression", "Pureish"],
   fields: {
-    params: {
-      validate: chain(assertValueType("array"), assertEach(assertNodeType("LVal"))),
+    // Unlike other functions, arrow functions are never generators.
+    ...(() => { const x = functionCommon; delete x.generator; return x; })(),
```

Comment:
There is a proposal for arrow function generators at Stage 1 (not implemented) https://github.com/tc39/proposals